File: brique-102/packages/php/src/Resources/Refunds.php
<?php

namespace Molam\Resources;

use Molam\Http\HttpClient;

class Refunds
{
    /** @var HttpClient */
    private $http;

    public function __construct(HttpClient $http)
    {
        $this->http = $http;
    }

    /**
     * Create a refund
     *
     * @param array $payload Refund data
     * @param string|null $idempotencyKey Optional idempotency key
     * @return array Refund response
     */
    public function create(array $payload, ?string $idempotencyKey = null): array
    {
        $options = [];
        if ($idempotencyKey) {
            $options['headers'] = ['Idempotency-Key' => $idempotencyKey];
        }

        $response = $this->http->post('/v1/refunds', [
            'refund' => $payload
        ], $options);

        return $response['data'] ?? [];
    }

    /**
     * Retrieve a refund
     */
    public function retrieve(string $id): array
    {
        $response = $this->http->get("/v1/refunds/{$id}");
        return $response['data'] ?? [];
    }

    /**
     * List refunds
     */
    public function list(array $params = []): array
    {
        $response = $this->http->get('/v1/refunds', ['query' => $params]);
        return $response['data'] ?? [];
    }
}

File: brique-102/packages/python/molam/resources/refunds.py
"""
Refunds resource
"""

from typing import Dict, Any, Optional, List


class Refunds:
    """Refunds API"""

    def __init__(self, http):
        self.http = http

    def create(
        self,
        payload: Dict[str, Any],
        idempotency_key: Optional[str] = None
    ) -> Dict[str, Any]:
        """
        Create a refund

        Args:
            payload: Refund data
            idempotency_key: Optional idempotency key

        Returns:
            Refund object
        """
        headers = {}
        if idempotency_key:
            headers["Idempotency-Key"] = idempotency_key

        response = self.http.post(
            "/v1/refunds",
            {"refund": payload},
            headers=headers if headers else None
        )
        return response.get("data", {})

    def retrieve(self, refund_id: str) -> Dict[str, Any]:
        """
        Retrieve a refund

        Args:
            refund_id: Refund ID

        Returns:
            Refund object
        """
        response = self.http.get(f"/v1/refunds/{refund_id}")
        return response.get("data", {})

    def list(self, params: Optional[Dict[str, Any]] = None) -> List[Dict[str, Any]]:
        """
        List refunds

        Args:
            params: Query parameters (limit, starting_after, etc.)

        Returns:
            List of refund objects
        """
        response = self.http.get("/v1/refunds", params=params)
        return response.get("data", [])
